pkg/api/v1alpha1/cluster_types.go

```diff
@@ -1485,6 +1486,16 @@ type ClusterList struct {
 	Items           []Cluster `json:"items"`
 }
 
+// ClusterNameIndexer is an indexer for controller to list Cluster objects based on name.
+func ClusterNameIndexer(obj client.Object) []string {
```

Comment:
can we put this somewhere else?
I would rather not have controller-runtime as a dependency in the api package
maybe `controller/clientutil`?

Reply:
Yea good idea, I wasn't happy with this location too